releases/models.py

```diff
@@ -313,3 +306,27 @@ def validate_artifact_name(self, name, suffix):
         regex = f"^[Dd]jango-{re.escape(version)}{re.escape(suffix)}$"
         message = f"Filename {name} does not match pattern {regex}."
         return RegexValidator(regex, message=message, code="invalid_name")(name)
+
+    def set_previous_release_as_eol(self):
+        """Handles setting EOL date for the previous release in the series."""
+        previous_release_kwargs = {
+            "major": self.major,
+            "minor": self.minor,
+            "micro": self.micro,
+            "status": self.status,
+            "eol_date__isnull": True,
+        }
+        if self.iteration > 1:
+            previous_release_kwargs["iteration"] = self.iteration - 1
+        elif self.status == "b":
+            previous_release_kwargs["status"] = "a"
+        elif self.status == "c":
+            previous_release_kwargs["status"] = "b"
+        elif self.status == "f" and self.micro == 0:
+            previous_release_kwargs["status"] = "c"
+        elif self.status == "f" and self.micro > 0:
+            previous_release_kwargs["micro"] = self.micro - 1
+
+        self.__class__.objects.filter(**previous_release_kwargs).update(
```

Comment:
I don't understand why we can't use the simpler form:
```suggestion
        self.filter(**previous_release_kwargs).update(
```

Reply:
Doh 🤦🏻  (for some reason I thought we were on the queryset, not the model)